src/types/did_parser.rs

```diff
@@ -29,53 +29,24 @@ peg::parser! {
         ///        }
         ///   });
         /// ```
-        pub rule ethr_did() -> MethodAndId
-            = method:method() id:id() { MethodAndId { method, id } }
+        pub rule ethr_did() -> Did
+            = method:method() ":" network:network()? account:account() { Did { method, network: network.unwrap_or_default(), account } }
 
         rule method() -> Method
-            = "ethr:" { Method::Ethr } / expected!("the only supported method is `ethr`")
+            = "ethr" { Method::Ethr } / expected!("the only supported method is `ethr`")
 
-        rule id() -> Id
-            = id: id_no_network() { id } / id:id_and_network() { id }
+        rule network() -> Network
```

Comment:
method identifier here

Reply:
"network" should be referred to as method_id

Reply:
According to the [did-ethr spec](https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md#method-specific-identifier):

method is always "ethr" for did-ethr.
id contains network and account:
- network is "mainnet", "sepolia" or "0x1"(chain id).
- account is address or public key.